lib/interpreters/periodic_visits.rb

```diff
@@ -212,12 +215,18 @@ def generate_vehicles(vrp)
 
           timewindows = [vehicle.timewindow || vehicle.sequence_timewindows].flatten
           if timewindows.empty?
-            new_vehicle = build_vehicle(vrp, vehicle, vehicle_day_index, rests_durations)
-            new_vehicle
+            build_vehicle(vrp, vehicle, vehicle_day_index, nil, rests_durations)
           else
-            timewindows.select{ |timewindow| timewindow.day_index.nil? || timewindow.day_index == vehicle_day_index % 7 }.collect{ |associated_timewindow|
-              new_vehicle = build_vehicle(vrp, vehicle, vehicle_day_index, rests_durations)
-              new_vehicle.timewindow = Models::Timewindow.create(start: associated_timewindow.start || 0, end: associated_timewindow.end || 86400)
+            timewindows.select{ |timewindow|
+              timewindow.day_index.nil? || timewindow.day_index == vehicle_day_index % 7
+            }.collect{ |associated_timewindow|
+              # FIXME : this function assumes there are no two timewindows with same day_index.
```

Comment:
@braktar I thought you already fixed this issue weeks ago, I must have misunderstood. Doesn't this issue of "having two vehicles with the same id" lead to an internal problem/bug ? If so, maybe we should just add a verification to the API, no?

Reply:
Oh ! Yes ! It comes in another PR (https://github.com/braktar/optimizer-api/pull/14/files) which requires : https://github.com/Mapotempo/optimizer-api/pull/353

Reply:
Let's keep this one (#273) for the end. First we need #353 and then [https://github.com/braktar/optimizer-api/pull/14/](https://github.com/braktar/optimizer-api/pull/14

Reply:
- [ x ] https://github.com/Mapotempo/optimizer-api/pull/353 is merged

Reply:
You can rebase & move 14 to this side

Reply:
Yes ! The FIXME has no sense with the vehicle name fix

Reply:
ping to remind rebase and the FIXME fix :)  then we can merge this.